modules/account-lib/src/coin/sol/transactionBuilder.ts

```diff
@@ -21,55 +38,210 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
+      const secretKey = signer.getKeys(true).prv;
+      assert(secretKey instanceof Uint8Array);
+      tx.partialSign({ publicKey, secretKey });
+    }
+    return tx;
+  }
 
   // region Getters and Setters
   /** @inheritdoc */
   protected get transaction(): Transaction {
     return this._transaction;
   }
+
+  /** @inheritdoc */
+  protected set transaction(transaction: Transaction) {
+    this._transaction = transaction;
+  }
+
+  /** @inheritdoc */
+  protected signImplementation(key: BaseKey): Transaction {
+    this.validateKey(key);
+    this.checkDuplicatedSigner(key);
+    const prv = key.key;
+    const signer = new KeyPair({ prv: prv });
+    this._signers.push(signer);
+
+    return this._transaction;
+  }
+
+  /**
+   * Set the transaction fee payer
+   *
+   * @param {string} feePayerAddress the fee payer account
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  feePayer(feePayerAddress: string | undefined): this {
+    if (!feePayerAddress || !isValidPublicKey(feePayerAddress)) {
+      throw new BuildTransactionError('Invalid or missing feePayerAddress, got: ' + feePayerAddress);
+    }
+    this._feePayer = feePayerAddress;
+    return this;
+  }
+
+  /**
+   * Set the transaction nonce
+   * Requires both optional params in order to use the durable nonce
+   *
+   * @param {Blockhash} blockHash The latest blockHash
+   * @param {boolean} useDurableNonce Set to true to use Durable Nonce, default is false
+   * @param {string} [walletNoncePubKey] the pub key of the nonce Wallet (required for durable nonce)
+   * @param {string} [authWalletPubKey] the pub key of the owner of the nonce Wallet (required for durable nonce)
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  nonce(blockHash: Blockhash, useDurableNonce = false, walletNoncePubKey?: string, authWalletPubKey?: string): this {
+    if (!blockHash || !isValidBlockId(blockHash)) {
+      throw new BuildTransactionError('Invalid or missing blockHash, got: ' + blockHash);
+    }
+    if (useDurableNonce) {
+      if (!walletNoncePubKey || !isValidPublicKey(walletNoncePubKey)) {
+        throw new BuildTransactionError('Invalid or missing walletNoncePubKey, got: ' + walletNoncePubKey);
+      }
+      if (!authWalletPubKey || !isValidPublicKey(authWalletPubKey)) {
+        throw new BuildTransactionError('Invalid or missing authWalletPubKey, got: ' + authWalletPubKey);
+      }
+      if (walletNoncePubKey === authWalletPubKey) {
+        throw new BuildTransactionError('Invalid params: walletNoncePubKey cannot be equal to authWalletPubKey');
+      }
+      this._nonceInfo = {
+        type: InstructionBuilderTypes.NonceAdvance,
+        params: { walletNoncePubKey, authWalletPubKey },
+      };
+    }
+    this._recentBlockhash = blockHash;
+    return this;
+  }
+
   // endregion
 
   // region Validators
   /** @inheritdoc */
   validateAddress(address: BaseAddress, addressFormat?: string): void {
```

Comment:
let's be consistent with type of `address` in builder funcs. the ones we added are all strings. we should probably update them to `BaseAddress`